scipy/stats/stats.py

```diff
@@ -4477,10 +4491,18 @@ def count_rank_tie(ranks):
     # Note that tot = con + dis + (xtie - ntie) + (ytie - ntie) + ntie
     #               = con + dis + xtie + ytie - ntie
     con_minus_dis = tot - xtie - ytie + ntie - 2 * dis
-    tau = con_minus_dis / np.sqrt(tot - xtie) / np.sqrt(tot - ytie)
+    if variant == 'b':
+        tau = con_minus_dis / np.sqrt(tot - xtie) / np.sqrt(tot - ytie)
+    elif variant == 'c':
+        minclasses = min(len(set(x)), len(set(y)))
+        tau = 2*con_minus_dis / (size**2 * (minclasses-1)/minclasses)
+    else:
+        raise ValueError("Unknown variant of the method chosen: "+str(variant)+". Please use 'b' or 'c'.")
+
     # Limit range to fix computational errors
     tau = min(1., max(-1., tau))
 
+    # The p-value calculation is the same for all variants since the p-value depends only on con_minus_dis
```

Comment:
PEP 8: line too long.
```suggestion
    # The p-value calculation is the same for all variants since the p-value
    # depends only on con_minus_dis.
```